### PR TITLE
check for security context CKV_K8S_29 for terraform k8s

### PR DESCRIPTION
**Pull request by JamesWoolfenden**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/terraform/checks/resource/kubernetes/PodSecurityContext.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+class PodSecurityContext(BaseResourceCheck):
+
+    def __init__(self):
+        # CIS-1.5 5.7.3
+        name = "Apply security context to your pods and containers"
+        # Security context can be set at pod or container level.
+        id = "CKV_K8S_29"
+
+        supported_resources = ['kubernetes_pod', 'kubernetes_deployment', 'kubernetes_daemonset']
+        categories = [CheckCategories.GENERAL_SECURITY]
```
Comment (rotemavni) [inline checkov/terraform/checks/resource/kubernetes/PodSecurityContext.py:14]: why not `CheckCategories.KUBERNETES`?

**Review by rotemavni — COMMENTED** [review]
Nice 😄 
Please refer to the comments in this PR - https://github.com/bridgecrewio/checkov/pull/2669
We need to be more careful when accessing attributes without validation (or catching possible errors)

File: checkov/terraform/checks/resource/kubernetes/PodSecurityContext.py (hunk 1 above)
Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/kubernetes/PodSecurityContext.py:14]: because every check for the provider would be that.

Comment (JamesWoolfenden) [reply checkov/terraform/checks/resource/kubernetes/PodSecurityContext.py:14]: Ive added type checking

**Pull request merged**